### ISSUE TITLE
Support for route error substate

### ISSUE THREAD

**Issue by rogatty** [ember-fastboot/fastboot#134]
## Expected behaviour
When the application route model rejects and the app enters an error substate, FastBoot should render a user-friendly error page and set a desired status code on the response.

## Actual behaviour
FastBoot throws an error and renders empty page with status code 503.

## Workaround
Enter the error substate manually and prevent bubbling the error up:
```
routes/application.js

actions: {
  error(err) {
    const fastboot = this.get('fastboot');

    if (fastboot.get('isFastBoot')) {
      fastboot.set('response.statusCode', err.code);
      this.intermediateTransitionTo('application_error', err);
      
      return false;
    }

    return true;
  }
}
```

**Comment by kratiahuja** [comment]
Just brainstorming: Can this not be handled in `fastboot-app-server` or a middleware? FastBoot library is app agnostic. It simply will return the serialized HTML and inject it into the base page.

**Comment by rogatty** [comment]
Might have been fixed by https://github.com/emberjs/ember.js/pull/15689, something to check.

**Comment by kratiahuja** [comment]
Going to close this as this is fixed in Ember 2.15.1. Please reopen if this is still an issue.
